src/main/java/com/example/qnacomunity/service/QnaService.java

```diff
@@ -0,0 +1,258 @@
+import com.example.qnacomunity.entity.Question;
+import com.example.qnacomunity.exception.CustomException;
+import com.example.qnacomunity.exception.ErrorCode;
+import com.example.qnacomunity.repository.AnswerRepository;
+import com.example.qnacomunity.repository.MemberRepository;
+import com.example.qnacomunity.repository.QuestionRepository;
+import java.time.LocalDateTime;
+import java.util.Objects;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@RequiredArgsConstructor
+public class QnaService {
+
+  private final QuestionRepository questionRepository;
+  private final AnswerRepository answerRepository;
+  private final MemberRepository memberRepository;
+
+  @Transactional
+  public QuestionResponse createQuestion(MemberResponse memberResponse, QuestionForm form) {
+
+    Member member = getMember(memberResponse);
+
+    Question question = Question.builder()
+        .member(member)
+        .title(form.getTitle())
+        .content(form.getContent())
+        .reward(form.getReward())
+        .build();
+
+    return QuestionResponse.from(questionRepository.save(question));
+  }
+
+  @Transactional
+  public QuestionResponse getQuestion(Long questionId) {
+
+    Question question = questionRepository.findById(questionId)
+        .orElseThrow(() -> new CustomException(ErrorCode.Q_NOT_FOUND));
+
+    //질문 조회수 1 증가
+    question.setHits(question.getHits() + 1);
+
+    return QuestionResponse.from(question);
+  }
+
+  @Transactional(readOnly = true)
+  public Page<QuestionResponse> getQuestions(Pageable pageable) {
+
+    //Questions 조회, Page 반환
+    Page<Question> questions = questionRepository.findAll(pageable);
+
+    return questions.map(QuestionResponse::from);
+  }
+
+  @Transactional(readOnly = true)
+  public Page<QuestionResponse> getMyQuestions(MemberResponse memberResponse, Pageable pageable) {
+
+    //해당 멤버가 생성한 Questions 최신순으로 반환
+    Page<Question> questions =
+        questionRepository.findAllByMember_Id(memberResponse.getId(), pageable);
+
+    return questions.map(QuestionResponse::from);
+  }
+
+  @Transactional
+  public QuestionResponse updateQuestion(
+      MemberResponse memberResponse,
+      Long questionId,
+      QuestionForm form
+  ) {
+
+    Question question = questionRepository.findById(questionId)
+        .orElseThrow(() -> new CustomException(ErrorCode.Q_NOT_FOUND));
+
+    //질문자 본인 맞는지 체크
+    if (!Objects.equals(question.getMember().getId(), memberResponse.getId())) {
+      throw new CustomException(ErrorCode.Q_MEMBER_NOT_MATCH);
+    }
+
+    //질문에 답변이 달려있는지
+    boolean isQuestionAnswered = answerRepository.existsByQuestion_Id(questionId);
+
+    //질문 달린 답변은 수정 불가
+    if (isQuestionAnswered) {
+      throw new CustomException(ErrorCode.Q_ANSWERED);
+    }
```

Comment:
isQuestionAnswered 의 경우는 별도로 선언해주지 않고 if 문안에서 바로 구해도 될것 같습니다

```suggestion

    //질문 달린 답변은 수정 불가
    if (answerRepository.existsByQuestion_Id(questionId)) {
      throw new CustomException(ErrorCode.Q_ANSWERED);
    }
```